# Marriott Hotel Category Change 2020 Analysis

In [2]:
# import pandas
import pandas as pd

In [3]:
# read data from csv
df = pd.read_csv('marriott-category-changes-2020.csv')
df.head()

Hotel                  Brand     Destination  \
0     Aberdeen Marriott Hotel               Marriott  United Kingdom   
1             AC Hotel Aitana  AC Hotels by Marriott           Spain   
2  AC Hotel Alcala de Henares  AC Hotels by Marriott           Spain   
3            AC Hotel Almeria  AC Hotels by Marriott           Spain   
4            AC Hotel Aravaca  AC Hotels by Marriott           Spain   

   Current Category  Current Standard Price  New Category  New Standard Price  
0                 3                   17500             2               12500  
1                 3                   17500             4               25000  
2                 1                    7500             2               12500  
3                 1                    7500             2               12500  
4                 1                    7500             2               12500

## Question 1
Discribe the data types for each feature/column, e.g., xxx feature's data type is String, yyy feature's data type is float, etc.

In [4]:
df.dtypes

Hotel                     object
Brand                     object
Destination               object
Current Category           int64
Current Standard Price     int64
New Category               int64
New Standard Price         int64
dtype: object

* The Hotel feature's data type is String
* The Brand feature's data type is String
* The Destination feature's data type is String
* The Current Category feature's data type is an integer
* The Current Standard Price feature's data type is integer
* The New Category feature's data type is an integer
* The New Standard Price feature's data type is integer

## Question 2
- How many hotels are in this dataset?
- The hotels are from how many unique brands?
- Which destination/country has the most hotels listed in this dataset? List the total number of hotels in that country
- How many brands in China have hotel category changes?

In [5]:
df.shape

(2185, 7)

In [6]:
df.drop_duplicates()

Hotel                  Brand  \
0                       Aberdeen Marriott Hotel               Marriott   
1                               AC Hotel Aitana  AC Hotels by Marriott   
2                    AC Hotel Alcala de Henares  AC Hotels by Marriott   
3                              AC Hotel Almeria  AC Hotels by Marriott   
4                              AC Hotel Aravaca  AC Hotels by Marriott   
...                                         ...                    ...   
2180  Xiamen Marriott Hotel & Conference Centre               Marriott   
2181        Xiangshui Bay Marriott Resort & Spa               Marriott   
2182                  Yogyakarta Marriott Hotel               Marriott   
2183            Zhejiang Taizhou Marriott Hotel               Marriott   
2184                     Zhuzhou Marriott Hotel               Marriott   

         Destination  Current Category  Current Standard Price  New Category  \
0     United Kingdom                 3                   17500             2   
1              Spain                 3                   17500             4   
2              Spain                 1                    7500             2   
3              Spain                 1                    7500             2   
4              Spain                 1                    7500             2   
...              ...               ...                     ...           ...   
2180           China                 2                   12500             3   
2181           China                 3                   17500             2   
2182       Indonesia                 2                   12500             3   
2183           China                 2                   12500             1   
2184           China                 2                   12500             1   

      New Standard Price  
0                  12500  
1                  25000  
2                  12500  
3                  12500  
4                  12500  
...                  ...  
2180               17500  
2181               12500  
2182               17500  
2183                7500  
2184                7500  

[2183 rows x 7 columns]

In [7]:
df.shape

(2185, 7)

There are 2185 hotels in this dataset.

In [5]:
df.Brand.nunique()

30

There are 30 unique brands of Marriott hotels.

In [38]:
df.Destination.value_counts().sort_values(ascending=False)

USA          1548
Canada         76
China          68
Spain          34
India          26
             ... 
Bahrain         1
Bolivia         1
Uganda          1
Hong Kong       1
Qatar           1
Name: Destination, Length: 98, dtype: int64

USA has the most hotels with 1548. 

In [7]:
df2= df[df.Destination == 'China']
df2= df2[df2['New Category'] != df2['Current Category']]
df2.Brand.nunique()

14

14 Brands in China changed category.

## Question 3
- What's the percentage of hotels worldwide with category upgrade in 2020?

In [8]:
df3= df[df['New Category'] > df['Current Category']]
df3.shape

(1686, 7)

In [9]:
percent_changed = round(df3.shape[0]/df.shape[0]*100,2)
print(percent_changed)

77.16


77.16% of hotels worldwide has category upgrades.

## Question 4
- List hotels with category changes greater than 1 if any, such as changing from category 3 to 5 or from category 7 to 4
- List all JW Marriott hotels in China that have a category upgrade

In [10]:
df['Change']= abs(df['New Category']-df['Current Category'])
df.head()

Hotel                  Brand     Destination  \
0     Aberdeen Marriott Hotel               Marriott  United Kingdom   
1             AC Hotel Aitana  AC Hotels by Marriott           Spain   
2  AC Hotel Alcala de Henares  AC Hotels by Marriott           Spain   
3            AC Hotel Almeria  AC Hotels by Marriott           Spain   
4            AC Hotel Aravaca  AC Hotels by Marriott           Spain   

   Current Category  Current Standard Price  New Category  New Standard Price  \
0                 3                   17500             2               12500   
1                 3                   17500             4               25000   
2                 1                    7500             2               12500   
3                 1                    7500             2               12500   
4                 1                    7500             2               12500   

   Change  
0       1  
1       1  
2       1  
3       1  
4       1

In [11]:
df4 = df[df.Change >= 2]
df4.head()

Hotel        Brand Destination  \
913  Four Points by Sheraton Bali, Ungasan  Four Points   Indonesia   

     Current Category  Current Standard Price  New Category  \
913                 4                   25000             2   

     New Standard Price  Change  
913               12500       2

The Four Points by Sheraton Bali, Ungasan in Indonesia was the only hotel to have a category change greater than 1. Falling from Category 4 to 2.

In [12]:
df5= df[df.Destination == 'China']
df5= df5[df5['Brand'] == 'JW Marriott']
df5= df5[df5['New Category']> df5['Current Category']]
df5.head()

Hotel        Brand Destination  \
1074          JW Marriott Hotel Chengdu  JW Marriott       China   
1078         JW Marriott Hotel Shenzhen  JW Marriott       China   
1079  JW Marriott Hotel Shenzhen Bao'an  JW Marriott       China   
1083        JW Marriott Hotel Zhengzhou  JW Marriott       China   

      Current Category  Current Standard Price  New Category  \
1074                 3                   17500             4   
1078                 3                   17500             4   
1079                 3                   17500             4   
1083                 2                   12500             3   

      New Standard Price  Change  
1074               25000       1  
1078               25000       1  
1079               25000       1  
1083               17500       1

In [13]:
df5.shape

(4, 8)

4 JW Marriott hotels in China that had a category upgrade.

## Question 5
Assume you are in Feburary 2020 and the category changes will take effect on March 4, 2020. You are planning your trip to Florence, Italy and Hong Kong, China in April. You only stay in category 8 hotel (existing category 8 or future category 8) and want to optimize your point spending. Based on the data, which hotel you should book? when should you book your hotels for Florence and Hong Kong? Why?

In [14]:
df6= df[df.Destination == 'Italy']
df7= df[df.Destination == 'China']
df6.head()

Hotel                  Brand  \
11                                    AC Hotel Brescia  AC Hotels by Marriott   
51                                    AC Hotel Venezia  AC Hotels by Marriott   
52                                    AC Hotel Vicenza  AC Hotels by Marriott   
145                 Cervo Hotel, Costa Smeralda Resort               Sheraton   
519  Cristallo, a Luxury Collection Resort & Spa, C...      Luxury Collection   

    Destination  Current Category  Current Standard Price  New Category  \
11        Italy                 1                    7500             2   
51        Italy                 5                   35000             6   
52        Italy                 1                    7500             2   
145       Italy                 7                   60000             8   
519       Italy                 8                   85000             7   

     New Standard Price  Change  
11                12500       1  
51                50000       1  
52                12500       1  
145               85000       1  
519               60000       1

In [15]:
df7.head()

Hotel      Brand Destination  Current Category  \
82            Aloft Guangzhou Tianhe      Aloft       China                 2   
83   Aloft Guangzhou University Park      Aloft       China                 1   
469        Courtyard Shanghai-Pudong  Courtyard       China                 2   
470           Courtyard Shenzhen Bay  Courtyard       China                 2   
484                 Courtyard Suzhou  Courtyard       China                 2   

     Current Standard Price  New Category  New Standard Price  Change  
82                    12500             3               17500       1  
83                     7500             2               12500       1  
469                   12500             3               17500       1  
470                   12500             3               17500       1  
484                   12500             3               17500       1

In [23]:
df8= pd.concat([df6,df7])

In [34]:
df9 = df8[df8['New Category']== 8]
df10 = df8[df8['Current Category']== 8]
df11 = pd.concat([df9,df10])

In [35]:
df11.head()

Hotel              Brand  \
145                  Cervo Hotel, Costa Smeralda Resort           Sheraton   
1968                     The Westin Excelsior, Florence             Westin   
519   Cristallo, a Luxury Collection Resort & Spa, C...  Luxury Collection   
2165                                        W Hong Kong           W Hotels   

     Destination  Current Category  Current Standard Price  New Category  \
145        Italy                 7                   60000             8   
1968       Italy                 7                   60000             8   
519        Italy                 8                   85000             7   
2165       China                 8                   85000             7   

      New Standard Price  Change  
145                85000       1  
1968               85000       1  
519                60000       1  
2165               60000       1

I should book my hotel for the W Hong Kong before the categories change on March 4th, but will have to wait to book the The Westin Excelsior, Florence until after that date.